File: app/src/main/java/com/sasfmlzr/findfilm/MainActivity.java
package com.sasfmlzr.findfilm;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.sasfmlzr.findfilm.fragment.CurrentFilmFragment;
import com.sasfmlzr.findfilm.fragment.ParentFilmFragment;
import com.sasfmlzr.findfilm.fragment.SettingsFragment;
import com.sasfmlzr.findfilm.model.SystemSettings;
import com.sasfmlzr.findfilm.service.NotificationService;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity implements ParentFilmFragment.filmClickedListener {
    private boolean startServiceOnDestroy;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        startServiceOnDestroy = true;
        setContentView(R.layout.activity_main);
        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);
        SharedPreferences.OnSharedPreferenceChangeListener prefListener = (prefs, key) -> {
            if (key.equals(SettingsFragment.KEY_LANGUAGE)) {
                startServiceOnDestroy = false;
                recreate();
            }
        };
        sharedPref.registerOnSharedPreferenceChangeListener(prefListener);

        SystemSettings.LANGUAGE = sharedPref.getString(SettingsFragment.KEY_LANGUAGE, "en_US");
        if (savedInstanceState == null) {
            createParentFragment();
        }
        //startService(new Intent(this, NotificationService.class));
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        //  if (startServiceOnDestroy) {
        //      startService(new Intent(this, NotificationService.class));
        //  }
    }

    @Override
    public void isClicked(int idFilm) {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.containerForFragment, CurrentFilmFragment.newInstance(idFilm))
                .addToBackStack(null)
                .commit();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.settings:
                setSettings();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    public void onBackPressed() {
        FragmentManager fm = getSupportFragmentManager();
        for (Fragment frag : fm.getFragments()) {
            if (frag.isVisible()) {
                FragmentManager childFm = frag.getChildFragmentManager();
                if (childFm.getBackStackEntryCount() > 0) {
                    childFm.popBackStack();
                    return;
                } else {
                    super.onBackPressed();
                }
            }
        }
    }

    private void createParentFragment() {
        getSupportFragmentManager().beginTransaction()
                .add(R.id.containerForFragment, new ParentFilmFragment())
                .commit();
    }

    private void setSettings() {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.containerForFragment, new SettingsFragment())
                .addToBackStack(null)
                .commit();
    }
}

File: app/src/main/java/com/sasfmlzr/findfilm/MainActivity.kt
package com.sasfmlzr.findfilm

import android.content.Intent
import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import android.os.Bundle
import android.preference.PreferenceManager
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.sasfmlzr.findfilm.fragment.currentfilm.CurrentFilmFragment
import com.sasfmlzr.findfilm.fragment.parentfilm.ParentFilmFragment
import com.sasfmlzr.findfilm.fragment.parentfilm.ParentFilmFragment.filmClickedListener
import com.sasfmlzr.findfilm.fragment.settings.SettingsFragment
import com.sasfmlzr.findfilm.model.SystemSettings
import com.sasfmlzr.findfilm.service.NotificationService

class MainActivity : AppCompatActivity(), filmClickedListener {
    private var startServiceOnDestroy = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        startServiceOnDestroy = true
        setContentView(R.layout.activity_main)
        val sharedPref = PreferenceManager.getDefaultSharedPreferences(this)
        prefListener = OnSharedPreferenceChangeListener { prefs: SharedPreferences?, key: String ->
            if (key == SettingsFragment.KEY_LANGUAGE) {
                startServiceOnDestroy = false
                recreate()
            }
        }
        sharedPref.registerOnSharedPreferenceChangeListener(prefListener)
        SystemSettings.LANGUAGE =
            sharedPref.getString(SettingsFragment.KEY_LANGUAGE, "en_US")
        if (savedInstanceState == null) {
            createParentFragment()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        if (startServiceOnDestroy) {
            startService(Intent(this, NotificationService::class.java))
        }
    }

    override fun isClicked(idFilm: Int) {
        supportFragmentManager.beginTransaction()
            .replace(R.id.containerForFragment, CurrentFilmFragment.newInstance(idFilm))
            .addToBackStack(null)
            .commit()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.settings -> {
                setSettings()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onBackPressed() {
        val fm = supportFragmentManager
        for (frag in fm.fragments) {
            if (frag.isVisible) {
                val childFm = frag.childFragmentManager
                if (childFm.backStackEntryCount > 0) {
                    childFm.popBackStack()
                    return
                } else {
                    super.onBackPressed()
                }
            }
        }
    }

    private fun createParentFragment() {
        supportFragmentManager.beginTransaction()
            .add(R.id.containerForFragment, ParentFilmFragment())
            .commit()
    }

    private fun setSettings() {
        supportFragmentManager.beginTransaction()
            .replace(R.id.containerForFragment, SettingsFragment())
            .addToBackStack(null)
            .commit()
    }

    companion object {
        var prefListener: OnSharedPreferenceChangeListener? = null
    }
}